src/project/project-actions.tsx

```diff
@@ -543,6 +596,10 @@ export class ProjectActions {
             showMultipleFilesHelp: false,
           });
         }
+        if (this.project.files.length === 1) {
+          // Saving the main file is an OK way to reset the dirty flag if there are no other files.
+          await this.fs.clearDirtyFlag();
+        }
```

Comment:
Rob points out there's some risk that the user won't actually have the file as browsers default to suggesting you discard .py file downloads. Perhaps we should just drop this bit.

Reply:
I don't have a strong view on this.

Reply:
This currently doesn't work as it needs moving out of it's immediate parent block. I find it strange that it asks for confirmation after saving main.py, so my preference is keeping this once fixed.

Reply:
Ooh, I meant to writ e2e for this and the regular Save reset but didn't.

Reply:
Scope fixed in a121783f2c6cc31258fb3779d5c6a245684c85e3

Reply:
e2e tests added too.